cmd/migration-managerd/internal/api/workers.go

```diff
@@ -1048,3 +1060,208 @@ func (d *Daemon) configureMigratedInstances(ctx context.Context, instances migra
 		return nil
 	})
 }
+
+func (d *Daemon) checkSourceConnectivity() {
+	log := slog.With(slog.String("method", "checkSourceConnectivity"))
+
+	// TODO: context should be passed from the daemon to all the workers.
+	ctx := context.TODO()
```

Comment:
On the call site of this we do have a context, so we should take it as an argument.

And while speaking of this, shouldn't this function have an `error` as return value?

Reply:
Updated to take context as an argument and also return an error.